pkg/kubelet/userns_manager.go

```diff
@@ -0,0 +1,476 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"os"
+	"path"
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+	"k8s.io/kubernetes/pkg/features"
+)
+
+const bitsUint32 = 32
+
+type bitArray struct {
```

Comment:
This is an odd name for a struct that _isn't_ an array.

Reply:
not sure, I've always called it bit array or bit map: https://en.wikipedia.org/wiki/Bit_array

Do you have a suggestion for a better name?

Reply:
@sftim What do you mean it is not an array? Can you please elaborate?

It is that, the `data` field is an array/slice of uint32. We are creating a bitmap or bitArray with that (see the wikipedia article @giuseppe linked).

Reply:
Let's leave it.

Reply:
I still don't see what was confusing for you, but sure. Marking it as solved